### DISCUSSION TITLE
Skeleton Loader

### DISCUSSION THREAD

**Discussion by char-mainewang** [twilio-labs/paste#2421] [Q&A] [answered]
Hey Paste team, are there any guidance for how we should use a [skeleton loader](https://paste.twilio.design/components/data-grid/#data-grid-in-a-loading-state) for tables where the content (number of rows and columns) is unpredictable until the data loads?

**Comment by SiTaggart** [comment, marked as answer]
Hi @char-mainewang, this was briefly touched upon in another discussion. It might not be the definitive answer, but little background knowledge to get started with. https://github.com/twilio-labs/paste/discussions/1932#discussioncomment-2717539

**Reply by nkrantz** [reply]
Hey @char-mainewang - the structure of the content displayed with the Skeleton Loader doesn't need to be an exact match to the data that will replace it. It's more so to give the user an _idea_ of what is going to be displayed. We'd recommend showing a [Skeleton Loader in a Table](https://paste.twilio.design/components/skeleton-loader#loading-a-table) even if you're not sure how many rows/columns there will be. It's ok to make an estimate - probably around 3-4 columns. You can also Skeleton Loader-fy the column headings like in the screenshot below since you're not sure what those will be.

<img width="1259" alt="image" src="https://user-images.githubusercontent.com/75342690/169567992-6fa4b94e-8ed8-49e1-858e-200eb1921c2b.png">

**Reply by char-mainewang** [reply]
Thanks y'all, this is super helpful!
